### PR TITLE
789 refactor auth namespace

### PR DESCRIPTION
**Pull request by Jesus89**
Related to https://github.com/CartoDB/cartoframes/issues/789.

This is a first step in the auth refactor.

- Move Credentials to auth
- Move CartoContext to auth and rename it to Context
- Update tests
- Update docs

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -221,9 +221,9 @@ The following will embed a CARTO map in a Jupyter notebook, allowing for custom
 .. code:: python
 
     from cartoframes import Layer, BaseMap, styling
-    cc = cartoframes.CartoContext(base_url=BASEURL,
+    c = cartoframes.auth.Context(base_url=BASEURL,
```
Comment (andy-esch) [inline README.rst]: How about renaming to `con` instead?

File: cartoframes/data/dataset.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def __init__(self, table_name=None, schema=None,
                  query=None, df=None, gdf=None,
                  state=None, is_saved_in_carto=False, context=None):
         from ..auth import _default_context
-        self._cc = context or _default_context
+        self._c = context or _default_context
```
Comment (andy-esch) [inline cartoframes/data/dataset.py]: I know `cc` wasn't the best name either, but how about `_con` here too? It improves readability if a user isn't familiar with the code base. I think pep8 will not like variable names like this too.

File: docs/etl.rst [hunk 3]
```diff
@@ -19,7 +19,7 @@ One common use case for cartoframes is its use in an ETL (Extract, Transform, an
     processed_data = <some processing pipeline>
 
     # Load into your carto account
-    cc.write(processed_data, 'processed_data')
+    c.data_boundaries(processed_data, 'processed_data')
```
Comment (andy-esch) [inline docs/etl.rst]: Should be `c.write`

**Comment by Jesus89** [comment]
Hi @andy-esch I have renamed the variable, so it's ready to another review :)

**Pull request merged**